packages/firestore/src/api/database.ts

```diff
@@ -1915,7 +1857,95 @@ export class Query implements firestore.Query {
     );
   }
 
-  private validateNewFilter(filter: Filter): void {
+  /**
+   * Verifies that the arguments passed in can create a valid filter.
+   */
+  private validateFilter(
+    fieldPath: FieldPath,
+    relationOp: RelationOp,
+    value: unknown
+  ): void {
+    if (fieldPath.isKeyField()) {
+      if (relationOp === RelationOp.ARRAY_CONTAINS) {
+        throw new FirestoreError(
+          Code.INVALID_ARGUMENT,
+          "Invalid Query. You can't perform array-contains queries on " +
+            'FieldPath.documentId() since document IDs are not arrays.'
+        );
+      }
+      if (typeof value === 'string') {
+        if (value === '') {
+          throw new FirestoreError(
+            Code.INVALID_ARGUMENT,
+            'Function Query.where() requires its third parameter to be a ' +
+              'valid document ID if the first parameter is ' +
+              'FieldPath.documentId(), but it was an empty string.'
+          );
+        }
+        if (
+          !this._query.isCollectionGroupQuery() &&
+          value.indexOf('/') !== -1
+        ) {
+          throw new FirestoreError(
+            Code.INVALID_ARGUMENT,
+            `Invalid third parameter to Query.where(). When querying a collection by ` +
+              `FieldPath.documentId(), the value provided must be a plain document ID, but ` +
+              `'${value}' contains a slash.`
+          );
+        }
+        const path = this._query.path.child(ResourcePath.fromString(value));
+        if (!DocumentKey.isDocumentKey(path)) {
+          throw new FirestoreError(
+            Code.INVALID_ARGUMENT,
+            `Invalid third parameter to Query.where(). When querying a collection group by ` +
+              `FieldPath.documentId(), the value provided must result in a valid document path, ` +
+              `but '${path}' is not because it has an odd number of segments (${
+                path.length
+              }).`
+          );
+        }
+      } else if (!(value instanceof DocumentReference)) {
+        throw new FirestoreError(
+          Code.INVALID_ARGUMENT,
+          `Function Query.where() requires its third parameter to be a ` +
+            `string or a DocumentReference if the first parameter is ` +
+            `FieldPath.documentId(), but it was: ` +
+            `${valueDescription(value)}.`
+        );
+      }
+    }
+  }
+
+  /**
+   * Given an unknown value, finds the field value.
+   */
+  private findFieldValue( fieldPath: FieldPath,
+    value: unknown): FieldValue {
```

Comment:
FWIW- This looks like it's not been formatted with prettier...  which is strange, since it's supposed to run automatically when you push... 🤔